planner/core/rule_column_pruning.go

```diff
@@ -145,8 +145,11 @@ func (ls *LogicalSort) PruneColumns(parentUsedCols []*expression.Column) error {
 				continue
 			}
 			ls.ByItems = append(ls.ByItems[:i], ls.ByItems[i+1:]...)
+		} else if ls.ByItems[i].Expr.GetType().Tp == mysql.TypeNull {
+			ls.ByItems = append(ls.ByItems[:i], ls.ByItems[i+1:]...)
 		} else {
 			parentUsedCols = append(parentUsedCols, cols...)
+
```

Comment:
Useless blank line?

Reply:
Oh forget to push the cleanup commit to github remote branch